src/main/scala/java/lang/ref/Reference.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * scalajs-weakreferences (https://github.com/scala-js/scala-js-weakreferences)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.lang.ref
+
+import scala.scalajs.js
+
+/* The JavaDoc says that the methods are concrete in `Reference`, and not
+ * overridden in `WeakReference`. To mimic this setup, and since
+ * `WeakReference` is the only implemented subclass, we actually implement the
+ * behavior of `WeakReference` in `Reference`.
+ */
+abstract class Reference[T] private[ref] (referent: T, queue: ReferenceQueue[_ >: T]) {
+  private[this] var weakRef = new js.WeakRef(referent)
```

Comment:
At the risk of being pedantic: Doesn't this line depend on unspecified behavior? Essentially, it requires that objects in Scala.js are implemented through JavaScript objects, right?

Do we have any provisions (e.g. in the versioning of this library), that will allow us to change object representation in SJS?